### ISSUE TITLE
Better POST support for conversation actions that require POST

### ISSUE THREAD

**Issue by hodgestar** [praekeltfoundation/vumi-go#708]
Currently all conversation action buttons use GET requests which is fine for actions with forms but doesn't do anything useful for actions that have no forms and require POST requests. We currently work around this by adding forms with no input fields which is decidedly non-optimal UI design.

**Comment by smn** [comment]
My _guess_ is that `send_survey` works because it's got an empty form defined. If that's the case then https://gist.github.com/smn/a4e552c8b21d41a2a5f0 could make it work for `download_user_data`.

**Comment by hodgestar** [comment]
Correct. I've created https://gist.github.com/hodgestar/6569689 with a temporary fix (I don't think that closes this ticket though). Also, there currently isn't any UI for custom router actions so those are okay.

**Comment by hodgestar** [comment]
https://gist.github.com/hodgestar/6569722 -- combined workaround gist.

**Comment by smn** [comment]
:+1: on applying the workaround but leaving this issue open for a proper fix.

**Comment by jerith** [comment]
Yes, the empty form is because we couldn't easily do `POST` actions in the UI at the time this code was originally written.
